src/App.js

```diff
@@ -44,29 +47,44 @@ const App = () => {
       );
   }, []);
 
-  function showConfirm(index) {
-    setDeleteUserIndex(index);
-    setShowModal(true);
+  function showDeleteConfirm(robotIndex) {
+    setDeleteRobotIndex(robotIndex);
+    setShowDeleteModal(true);
   }
 
   function deleteItem() {
-    setRobotList(robotList.filter((item, index) => index !== deleteUserIndex));
-    setShowModal(false);
+    setRobotList(robotList.filter((item, index) => index !== deleteRobotIndex));
+    setShowDeleteModal(false);
+  }
+
+  function showExtraInformation(robotInformationIndex) {
+    setExtraRobotInformationIndex(robotInformationIndex)
+    setShowRobotModal(true)
   }
 
   return (
     <div>
       {
         /* Only render modal if items are loaded */
         // main takeaway for this PR is, name your variables more specifically, helps with debugging
-        robotList.length > 0 && showModal && (
-          <NewModal
+        robotList.length > 0 && showDeleteModal && (
+          <DeleteModal
             onDo={deleteItem}
-            onClose={() => setShowModal(false)}
-            user={robotList[deleteUserIndex].name}
+            onClose={() => setShowDeleteModal(false)}
+            RobotNameToBeDeleted={robotList[deleteRobotIndex].name}
           />
         )
       }
+      {robotList.length > 0 && showRobotModal && (
```

Comment:
Since there's a few things that need `robotList` to not be empty, maybe you could wrap all of it under one `if` condition instead of checking it multiple times

Reply:
Did a ternary statement